Make updates relating to communication channels. (#938)
python/devguide: developer-workflow/communication-channels.rst

Mailing Lists
=============

.. note:: Some mailing lists have been supplanted by categories in the

>   Python `Discourse`_. Specifically,

Reviewer: ```suggestion
   Python `Discourse`_. Specifically,
```

I think it was my mistake, but this missing space is inserting an extra quote level

> .. note:: Some mailing lists have been supplanted by categories in the `discussion
>    forum <https://discuss.python.org/>`_. Specifically,

Reviewer: ```suggestion
.. note:: Some mailing lists have been supplanted by categories in the
  Python `Discourse`_. Specifically,
```

* Use the existing link instead of repeating it inline
* More precise/specific and commonly used name

   * The python-dev mailing list is supplanted by posts in the `Core Development`_
     and `PEPs`_ categories on the forum.

   * The python-ideas mailing list is supplanted by posts in the `Ideas`_

>      category on the forum.

Author: Should be "superseded by"

>      category on the forum.

Reviewer: Oops, silly typo sorry

   Discussion in ``Core Development`` is focused on issues related to Python's
   development, such as how to handle a specific issue, a PEP, etc.

>    - Ideas about new functionality should **not** start here and instead

Reviewer: ```suggestion
   - Ideas about new functionality should **not** start here, and instead
```

Missing comma

     should be discussed in `Ideas`_.

>    - Technical support questions should also not be asked here and instead

Reviewer: ```suggestion
   - Technical support questions should also not be asked here, and instead
```

Missing comma

     should go to the python-list_ or python-help_ mailing lists, or the

>      `Python Help`_ category on the forum.

Reviewer: ```suggestion
     `Python Help`_ category on Discourse.
```

Clearer, more common and more specific

> Old threads in the python-dev_, python-committers_, and python-ideas_ mailing lists

Reviewer: Also, why not link the archives directly here instead of repeating a link to the list signup page?

> can be accessed through the Mailman 3 `web gateway`_.

Reviewer: ```suggestion
can be accessed through the `online archive <web gateway_>`__.
```

Maybe a more appropriate term than bringing up an implementation detail?

General Python questions should go to `python-list`_ or `tutor`_

> or similar resources, such as StackOverflow_ or the ``#python`` IRC channel

Reviewer: I can't reach it, but it might be worth updating the links in the "complete list of Python mailing lists" since it still points to the Mailman 2 instance, when all but a few active lists have been migrated to Mailman 3 by now.

Author: I'll leave the Mailman2 link there, but add the Mailman3 link.

Mailing Lists
=============

.. note:: Some mailing lists have been supplanted by categories in the `discussion
   forum <https://discuss.python.org/>`_. Specifically,

   * The python-dev mailing list is supplanted by posts in the ``Core Development``
     and ``PEPs`` categories on the forum.

   * The python-ideas mailing list is supplanted by posts in the ``Ideas``
     category on the forum.

   Discussion in ``Core Development`` is focused on issues related to Python's
   development, such as how to handle a specific issue, a PEP, etc.

   - Ideas about new functionality should **not** start here and instead
     should be discussed in `Ideas`.
   - Technical support questions should also not be asked here and instead
     should go to the python-list_ or python-help_ mailing lists, or the
     ``Help`` category on the forum.

> For looking at old posts on the mailing lists, python-dev_, python-committers_, and
> python-ideas_ all use Mailman 3, and are hence accessible via the Mailman 3 `web
> gateway`_.

Reviewer 2: ```suggestion
Old threads in the python-dev_, python-committers_, and python-ideas_ mailing lists
can be accessed through the Mailman 3 `web gateway`_.
```

The python-checkins_ mailing list sends out an email for every commit to Python's

> various repositories from https://github.com/python/cpython. Many core developers

Reviewer 2: Do you mean `various branches`?  Mentioning `every commit to the python/cpython repository` should be enough.

Author: Yes, this was a typo.

subscribe to this list and are known to respond to these emails to make comments about

> various issues they catch in the commit. Replies get redirected to python-dev, but a

Reviewer 2: By looking at `python-checkins`, I don't really see any replies from core devs. By looking at emails sent to both `python-dev` and `python-checkins`, the most recent one is from 3 years ago.  ISTM that since we moved to GitHub everything is handled there and `python-checkins` is largely ignored (as you mentioned afterwards).

Author: Should `python-checkins` no longer be mentioned now, as it's of limited use to new committers / potential committers?

Reviewer 3: What is the value of mentioning `python-checkins`? I don't see any value of mentioning it here.

Author: I've removed that mention.

issue tracker is the place to discuss and work on improvements to the CPython
core development workflow.

> A complete list of Python mailing lists can be found at
> https://mail.python.org/mailman/listinfo (older lists, using Mailman2) or
> https://mail.python.org/archives/ (newer lists, using Mailman3). Some lists may also

Reviewer: Actually, these aren't _quite_ the same—the existing [mailman(2) link](https://mail.python.org/mailman/listinfo) points to the listing of list _info_ pages, e.g. [pages like this](https://mail.python.org/mailman/listinfo/python-list) (i.e. , whereas the [mailman3 link](https://mail.python.org/archives/) points to the list of Hyperkitty list _archives_, e.g. [pages like this](https://mail.python.org/archives/list/python-dev@python.org/), while the equivalent of the former for mailman3 is https://mail.python.org/mailman3/, i.e. currently linked by the target ``web gateway`` (AFAIK, there's no direct equivalent for Pipermail, other than, well, [this](https://mail.python.org/pipermail/).

So I suggest perhaps:

```suggestion
A complete list of Python mailing lists can be found at
https://mail.python.org/mailman/listinfo (older lists, using Mailman2) or
is https://mail.python.org/mailman3/ (newer lists, using Mailman3).
Some lists may also
```

> https://mail.python.org/archives/ (newer lists, using Mailman3). Some lists may also

Author: ```suggestion
https://mail.python.org/mailman3/ (newer lists, using Mailman3). Some lists may also
```
